From 09a010c86d223b89fe601eb07ca281e436ead7bb Mon Sep 17 00:00:00 2001
Message-Id: <09a010c86d223b89fe601eb07ca281e436ead7bb.1652216438.git.stefan@agner.ch>
In-Reply-To: <22e0fba77fffa3c5addabda4fd159a943ad7400f.1652216438.git.stefan@agner.ch>
References: <22e0fba77fffa3c5addabda4fd159a943ad7400f.1652216438.git.stefan@agner.ch>
From: Stefan Agner <stefan@agner.ch>
Date: Tue, 10 May 2022 22:57:26 +0200
Subject: [PATCH] ARM: dts: rockchip: Add Bluetooth to rk3288-tinker

Enable Bluetooth support via UART0. Use the existing bindings
for Realtek rtl8723bs Bluetooth device.

Signed-off-by: Stefan Agner <stefan@agner.ch>
---
 arch/arm/boot/dts/rk3288-tinker.dtsi | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/arch/arm/boot/dts/rk3288-tinker.dtsi b/arch/arm/boot/dts/rk3288-tinker.dtsi
index cdff130d1292..62443a2d020f 100644
--- a/arch/arm/boot/dts/rk3288-tinker.dtsi
+++ b/arch/arm/boot/dts/rk3288-tinker.dtsi
@@ -520,6 +520,16 @@ &tsadc {
 
 &uart0 {
 	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&uart0_xfer>, <&uart0_cts>, <&uart0_rts>;
+
+	bluetooth {
+		compatible = "realtek,rtl8723bs-bt";
+		device-wake-gpios = <&gpio4 26 GPIO_ACTIVE_HIGH>;
+		host-wake-gpios = <&gpio4 31 GPIO_ACTIVE_HIGH>;
+		enable-gpios = <&gpio4 29 GPIO_ACTIVE_HIGH>;
+		max-speed = <1500000>;
+	};
 };
 
 &uart1 {
-- 
2.36.1

